instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs

```diff
@@ -0,0 +1,110 @@
+-- option for module re-export
+{-# OPTIONS_GHC -Wno-missing-import-lists #-}
+
+-- | Wrapper module for @Database.Persist.MySQL@ with @OpenTelemetry.Instrumentation.Persistent@.
+module OpenTelemetry.Instrumentation.Persistent.MySQL (
+  withMySQLPool,
+  withMySQLConn,
+  createMySQLPool,
+  module Database.Persist.Sql,
+  MySQL.ConnectInfo (..),
+  MySQLBase.SSLInfo (..),
+  MySQL.defaultConnectInfo,
+  MySQLBase.defaultSSLInfo,
+  Orig.MySQLConf (..),
+
+  -- * @ON DUPLICATE KEY UPDATE@ Functionality
+  Orig.insertOnDuplicateKeyUpdate,
+  Orig.insertManyOnDuplicateKeyUpdate,
+  Orig.HandleUpdateCollision,
+  Orig.copyField,
+  Orig.copyUnlessNull,
+  Orig.copyUnlessEmpty,
+  Orig.copyUnlessEq,
+  openMySQLConn,
+) where
+
+import Control.Monad.IO.Unlift (MonadUnliftIO)
+import Control.Monad.Logger (MonadLoggerIO)
+import Data.Pool (Pool)
+import Data.Text (Text)
+import qualified Database.MySQL.Base as MySQLBase
+import qualified Database.MySQL.Simple as MySQL
+import qualified Database.Persist.MySQL as Orig
+import Database.Persist.Sql
+import qualified OpenTelemetry.Instrumentation.Persistent as Otel
+import qualified OpenTelemetry.Trace.Core as Otel
+
+
+{- | Create a MySQL connection pool.  Note that it's your
+responsibility to properly close the connection pool when
+unneeded.  Use 'withMySQLPool' for automatic resource control.
+-}
+createMySQLPool ::
+  (MonadUnliftIO m, MonadLoggerIO m) =>
+  Otel.TracerProvider ->
+  -- | Attributes that are specific to providers like MySQL, PostgreSQL, etc.
+  [(Text, Otel.Attribute)] ->
+  -- | Connection information.
+  MySQL.ConnectInfo ->
+  -- | Number of connections to be kept open in the pool.
+  Int ->
+  m (Pool SqlBackend)
```

Comment:
I know that `hs-opentelemetry-instrumentation-persistent` currently only supports `SqlBackend`, but I wonder if we could add support for `IsSqlBackend backend` as part of this PR?

Reply:
How should I change after all?